utf8.c

```diff
@@ -1099,6 +1100,15 @@ sequence.  You can use that function or C<L</utf8_to_uv_flags>> to exert more
 control over the input that is considered acceptable, and the warnings that are
 raised.
 
+So far unmentioned is a fifth function, C<utf8_to_uv_or_die>, suitable for use
+only in situations where it would be a bug in the program for the input to be
+malformed; typically it has already been validated.  The function returns the
+code point the UTF-8 represents instead of a boolean success or failure, and
+hence is lacking the parameter indicating where to store that code point.  That
+it returns at all indicates success, since it causes the Perl interpreter to
+croak upon failure.  Messages are displayed as to the cause of the croak even
+if warnings are off.
```

Comment:
Does it display messages if `$SIG{__DIE__}` is set?

If it's just the croak expected to display the exception, you could just say it throws an exception describing any errors found.

I do think this could be terser, there's no need to mention that it's unmentioned, nor that it is the fifth function, maybe something like:

```
The C<utf8_to_uv_or_die> has a simpler interface, for use when any errors are fatal.  It returns the code point instead of using an output parameter, and throws an exception with any errors found where the other functions here would have returned false.
```

Reply:
Your wording is much better